app/src/main/java/net/squanchy/navigation/HomeActivity.java

```diff
@@ -1,46 +1,77 @@
+import android.graphics.drawable.ColorDrawable;
 import android.os.Bundle;
+import android.support.annotation.AttrRes;
+import android.support.annotation.ColorInt;
 import android.support.annotation.Nullable;
 import android.support.design.widget.BottomNavigationView;
+import android.support.transition.Fade;
+import android.support.transition.TransitionManager;
+import android.util.TypedValue;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+
+import java.util.HashMap;
+import java.util.Map;
 
 import net.squanchy.R;
 import net.squanchy.fonts.TypefaceStyleableActivity;
-import net.squanchy.navigation.view.HomeViewPagerAdapter;
-import net.squanchy.navigation.view.NoSwipeViewPager;
 
 public class HomeActivity extends TypefaceStyleableActivity {
 
-    private static final int KEEP_ALL_PAGES = 4;       // This is more than what we need but it's to have some leeway
+    private final Map<BottomNavigationSection, View> pageViews = new HashMap<>(4);
+
+    private int pageFadeDurationMillis;
+
+    private BottomNavigationSection currentSection;
+    private BottomNavigationView bottomNavigationView;
+    private ViewGroup pageContainer;
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_home);
 
-        BottomNavigationView bottomNavigationView = (BottomNavigationView) findViewById(R.id.bottom_navigation);
+        pageFadeDurationMillis = getResources().getInteger(android.R.integer.config_mediumAnimTime);
+
+        pageContainer = (ViewGroup) findViewById(R.id.page_container);
+        collectPageViewsInto(pageViews);
 
-        NoSwipeViewPager viewPager = (NoSwipeViewPager) findViewById(R.id.main_view_pager);
-        HomeViewPagerAdapter adapter = new HomeViewPagerAdapter(this);
-        viewPager.setAdapter(adapter);
-        viewPager.setOffscreenPageLimit(KEEP_ALL_PAGES);
+        bottomNavigationView = (BottomNavigationView) findViewById(R.id.bottom_navigation);
+        setupBottomNavigation(bottomNavigationView);
 
+        selectPage(BottomNavigationSection.SCHEDULE);
+    }
+
+    private void collectPageViewsInto(Map<BottomNavigationSection, View> pageViews) {
+        pageViews.put(BottomNavigationSection.SCHEDULE, pageContainer.findViewById(R.id.schedule_content_root));
+        pageViews.put(BottomNavigationSection.FAVORITES, pageContainer.findViewById(R.id.favorites_content_root));
+        pageViews.put(BottomNavigationSection.TWEETS, pageContainer.findViewById(R.id.tweets_content_root));
+        pageViews.put(BottomNavigationSection.VENUE_INFO, pageContainer.findViewById(R.id.venue_content_root));
+    }
+
+    private void setupBottomNavigation(BottomNavigationView bottomNavigationView) {
         BottomNavigationHelper.disableShiftMode(bottomNavigationView);
+        bottomNavigationView.setBackground(bottomNavigationView.getBackground().mutate());
```

Comment:
Because drawables are cached when retrieved from the resources, and we were sharing the `?colorPrimary` drawable which caused the weird issues we were seeing. When you change the colour on one side, all drawable instances change their colour. Mutating is basically cloning the drawable with a new, distinct state, which is what we needed. This will become obsolete with the circular reveals anyway as we'll wrap the existing drawable.